pkg/controller/common/webhook/webhook.go

```diff
@@ -101,18 +117,30 @@ func (v *validatingWebhook) Handle(ctx context.Context, req admission.Request) a
 			whlog.Error(err, "decoding old object from webhook request into type (%T)", oldObj)
 			return admission.Errored(http.StatusBadRequest, err).WithWarnings(warnings...)
 		}
-		_, err = obj.ValidateUpdate(oldObj)
-		if err != nil {
-			return admission.Denied(err.Error()).WithWarnings(warnings...)
-		}
+		warnings, err = obj.ValidateUpdate(oldObj)
 	}
 
 	if req.Operation == admissionv1.Delete {
-		_, err = obj.ValidateDelete()
-		if err != nil {
-			return admission.Denied(err.Error())
+		warnings, err = obj.ValidateDelete()
+	}
+	if err != nil {
+		var apiStatus apierrors.APIStatus
+		if errors.As(err, &apiStatus) {
+			return validationResponseFromStatus(false, apiStatus.Status()).WithWarnings(warnings...)
 		}
```

Comment:
I wonder if this change may have an impact or not for ECK users. IIUC, it may change the response code which was always 403 until now. This may be something to highlight in the release notes.